js/modules/k6/experimental/tracing/module.go

```diff
@@ -69,3 +76,54 @@ func (mi *ModuleInstance) newClient(cc goja.ConstructorCall) *goja.Object {
 
 	return rt.ToValue(NewClient(mi.vu, opts)).ToObject(rt)
 }
+
+// InstrumentHTTP instruments the HTTP module with tracing headers.
+//
+// When used in the context of a k6 script, it will automatically replace
+// the imported http module's methods with instrumented ones.
+func (mi *ModuleInstance) instrumentHTTP(options options) {
+	rt := mi.vu.Runtime()
+
+	// Initialize the tracing module's instance default client,
+	// and configure it using the user-supplied set of options.
+	mi.Client = NewClient(mi.vu, options)
+
+	// Explicitly inject the http module in the VU's runtime.
+	// This allows us to later on override the http module's methods
```

Comment:
nitpick: this doesn't prevent someone from calling `instrumentHTTP` multiple times with different settings.

I am not certain what the solution should be here. I would argue changing the client somehow instead of stacking them. But it is probably okay if we just return error for now :shrug:

Reply:
I agree with you, I would find it more convenient and explicit if we threw an error if calling `instrumentHTTP` a second time. We could even make the error explicit along the lines of "it looks like you're attempting to reconfigure the HTTP module's instrumentation on the fly. This is not supported. Consider using the tracing.Client instead".